### DISCUSSION TITLE
Mono / .NET based Program crashing, won't start (Noob problem/not bug)

### DISCUSSION THREAD

**Discussion by FaRuffo** [mono/mono#21800] [Q&A]
System: Artix Linux (Arch fork)
Mono: Installed package Mono 6.12

This is not a bug, but I'm seeking help as a noob with mono. I downloaded the Linux deb package of dwsim, which is based on .NET and written mostly in C#. I then ran it and got these error messages:

```
Mono: DllImport unable to load library 'libgtk-x11-2.0.so.0'.
libgtk-x11-2.0.so.0 assembly:<unknown assembly> type:<unknown type> member:(null)
at (wrapper managed-to-native) Gtk.Application.gtk_init_check(int&,intptr&)  (more lines deleted)

```
I then ran ldconfig:

```
# ldconfig -p|grep libgtk-x11-2.0.so.0
#libgtk-x11-2.0.so.0 (libc6,x86-64) => /usr/lib/libgtk-x11-2.0.so.0

```
Gut feel is that I have some config I need to do.

Thanks for your help.

**Comment by lextm** [comment]
That dwsim app was too old and relies on the old GTK bits modern Linux distributions are not shipping with.

If the app developers have new releases that match the latest Linux distributions, you should upgrade. And if no compatible new releases are there, you will have to hack the code base on your own, or move to an alternative app.
